pep-0710.rst

```diff
@@ -171,16 +171,146 @@ from the installer's cache.
+Following the :ref:`packaging:recording-installed-packages` specification,
+installers may keep additional installer-specific files in the ``.dist-info``
+directory.  To make sure this PEP does not cause any backwards compatibility
+issues, there was conducted a research whether introducing the
+``provenance_url.json`` file in the ``.dist-info`` directory is feasible and
+does not cause any backwards compatibility issues in the Python packaging
+ecosystem.
+
+A specification for :ref:`binary-distribution-format` lists files that can be
+present in the ``.dist-info`` directory. None of these file names collide with
+the proposed ``provenance_url.json`` file from this PEP.
+
+Presence of provenance_url.json in installers and libraries
+-----------------------------------------------------------
+
+A research conducted on the existing installers, libraries, and dependency
+managers in the Python ecosystem shown below proved there are no backwards
+compatibility issues and introducing the ``provenance_url.json`` file will not
+clash with any existing tool at the time of writing this PEP.
+
+pip
+~~~
+
+The function from pip's internal API responsible for installing wheels, named
+`_install_wheel
+<https://github.com/pypa/pip/blob/10d9cbc601e5cadc45163452b1bc463d8ad2c1f7/src/pip/_internal/operations/install/wheel.py#L432>`__,
+does not store any ``provenance_url.json`` file in the ``.dist-info``
+directory. Additionally, a prototype introducing the mentioned file to pip in
+`pypa/pip#11865`_ demonstrates incorporating logic for handling the
+``provenance_url.json`` file in pip's source code.
+
+As pip is used by some of the tools mentioned below to install Python package
+distributions, findings for pip apply to these tools as well as pip does not
+allow parametrizing creation of files in the ``.dist-info`` directory in its
+internal API. Most of the tools mentioned below that use pip invoke pip as a
+subprocess which has no effect on the eventual presence of the
+``provenance_url.json`` file in the ``.dist-info`` directory.
+
+distlib
+~~~~~~~
+
+`distlib <distlib_homepage_>`_ implements low-level functionality that manipulates with the
+``dist-info`` directory. The database of installed distributions does not use
+any file named ``provenance_url.json`` based on `the distlib's source code
+<https://github.com/pypa/distlib/blob/05375908c1b2d6b0e74bdeb574569d3609db9f56/distlib/database.py#L39-L40>`__.
+
+Pipenv
+~~~~~~
+
+`Pipenv <pipenv_homepage_>`_ uses pip `to install Python package distributions
+<https://github.com/pypa/pipenv/blob/babd428d8ee3c5caeb818d746f715c02f338839b/pipenv/routines/install.py#L262>`__
+. There wasn't identified any additional logic that would cause backwards
+compatibility issues when introducing the ``provenance_url.json`` file in the
+``.dist-info`` directory.
+
+installer
+~~~~~~~~~
+
+installer does not create ``provenance_url.json`` file explicitly.
+Nevertheless, as per :ref:`packaging:recording-installed-packages`
```

Comment:
```suggestion
Nevertheless, as per the :ref:`Recording Installed Projects <packaging:recording-installed-packages>`
```

* Add missing definite article
* Hardcode name to avoid relying on it making sense in context